src/kbmod/search/stack_search.h

```diff
@@ -82,6 +85,9 @@ class StackSearch {
 
     // Results from the grid search.
     TrajectoryList results;
+
+    // Trajectories that are being searched.
+    TrajectoryList gpu_search_list;
```

Comment:
We could expand this object variable to be used in other places by creating a helper function `set_search_list()` that sets `gpu_search_list` from a given vector. We could then use that function in both `prepare_batch_search()` and `search()` (instead of allocating a local `gpu_search_list` within the function). I think this would be cleaner from a naming perspective.

Reply:
I think with the new paradigm of `search_all` this is less of a problem.  In terms of having a `set_search_list()` I worry that if it's not explicitly asked for in `prepare_search` and `search_all` you'll find yourself calling them before setting the search list. We can throw an error in that case, but I think it might be a little more intuitive for it to be explicitly asked for in the functions. I don't feel particularly strong though.

Reply:
I think with the new paradigm of `search_all` this is less of a problem.  In terms of having a `set_search_list()` I worry that if it's not explicitly asked for in `prepare_search` and `search_all` you'll find yourself calling them before setting the search list. We can throw an error in that case, but I think it might be a little more intuitive for it to be explicitly asked for in the functions. I don't feel particularly strong though.

Reply:
Sounds good. Let's keep as-is.